### ISSUE TITLE
Parent menu item filter

### ISSUE THREAD

**Issue by tonypartridge** [joomla/joomla-cms#22271]
so the Menu Items list, parent filter if you have a 3 tired menu i.e.

Services
-Home Improvements
--Bathrooms
--Lounge
-House Cleaning
--Carpets
--Dusting

If you filter by parent menu item 'Services' you do not get the sub items.  i.e. bathrooms, lounge, carpets, dusting. 

Before I do any work to improve this, is this as the majority would expect? Or would you expect to on show what is directly under the parent?

**Comment by ReLater** [comment]
Hard to say what's better. It depends on the number of sub item levels. The best solution for me would be if filter "Select Max Levels" would start at the level of the selected parent menu item.

**Comment by brianteeman** [comment]
There is a use case for both options.
I am creating a dropdown menu structure -> I want to see all the level 1 menu items
I am working on one of those dropdowns and I want to see the parent and its children

> The best solution for me would be if filter "Select Max Levels" would start at the level of the selected parent menu item.

We dont have a way to select it :(

**Comment by tonypartridge** [comment]
So there is a bug, the max levels should still be taking effect within the parent filter and it's not. (I always miss that option)

**Comment by brianteeman** [comment]
Sorry - I forgot there was a parent menu item select now

**Comment by ReLater** [comment]
At the moment (3.9) it's like this: If a parent item is selected that is in level=1 or 2 and you set max. levels = 1 you get an empty list. I think also an empty list if you set it to 2. I'm too confused atm ;-)

**Comment by tonypartridge** [comment]
I’ll see what I can do, I did add the parent filter after all. I just forgot about max levels 😬😬
  

  
  

  
  
>

**Comment by ggppdk** [comment]
If i understand the issue 

we should have

Existing code
```php
		// Filter the items over the parent id if set.
		$parentId = $this->getState('filter.parent_id');

		if (!empty($parentId))
		{
			$query->where('a.parent_id = ' . (int) $parentId);
		}

		// ...
		// ... this code will not change
		// ...

		// Filter on the level.
		if ($level = $this->getState('filter.level'))
		{
			$query->where('a.level <= ' . (int) $level);
		}
```

New code like:
```php
		// Filter the items over the parent id if set.
		$parentId = $this->getState('filter.parent_id');

		if (!empty($parentId))
		{
			$level = $this->getState('filter.level');

			// Create a subquery for the sub-items list
			$subQuery = $db->getQuery(true)
				->select('sub.id')
				->from('#__menu as sub')
				->join('INNER', '#__menu as this ON sub.lft > this.lft AND sub.rgt < this.rgt')
				->where('this.id = ' . (int) $parentId);

			if ($level)
			{
				$subQuery->where('sub.level <= this.level + ' . (int) $level);
			}

			// Add the subquery to the main query
			$query->where('(a.parent_id = ' . (int) $parentId . ' OR a.parent_id IN (' . (string) $subQuery . '))');
		}

		// Filter on the level.
		elseif ($level = $this->getState('filter.level'))
		{
			$query->where('a.level <= ' . (int) $level);
		}
```

**Comment by ggppdk** [comment]
@tonypartridge 

i have made a PR,
thanks for bringing this up,
current behavior is a limitation that makes menu-item management harder

**Comment by ghost** [comment]
closed as having Pull Request #22319
